src/midiEditorCommands.cpp

```diff
@@ -1121,6 +1121,59 @@ void postMidiChangePitch(int command) {
 	outputMessage(s);
 }
 
+void postMidiMoveStart(int command) {
+	HWND editor = MIDIEditor_GetActive();
+	MediaItem_Take* take = MIDIEditor_GetTake(editor);	
+	// Get selected notes.
+	vector<MidiNote> selectedNotes = getSelectedNotes(take);
+	auto count = selectedNotes.size();
+	if (count == 0) {
+		return;
+	}
+	auto firstStart = selectedNotes.cbegin()->start;
+	bool generalize = count >= 8 || any_of(
+		selectedNotes.begin(), selectedNotes.end(),
+		[firstStart](MidiNote n) { return firstStart != n.start; }
+		);
+	if (!generalize) {
+		previewNotes(take, selectedNotes);
+	}
+	if (shouldReportNotes) {
+		ostringstream s;
+		if (generalize) {
+			s << count << (count == 1 ? " note" : " notes ");
+			switch (command) {
+				case 40181:
+					s << "pixel left";
+					break;
+				case 40182:
+					s << "pixel right";
+					break;
+				case 40183:
+					s << "grid unit left";
+					break;
+				case 40184:
+					s << "grid unit right";
+					break;				
+				default:
+					s << "start moved";
+					break;
+			}
+		} else{ 
+			for (auto note = selectedNotes.cbegin(); note != selectedNotes.cend(); ++note) {
+				if (note == selectedNotes.cbegin()) {
+					s << formatTime(note->start, TF_MEASURE) << " ";
```

Comment:
Can we do this using `firstStart` (defined earlier) before the loop, rather than doing this within the loop?

Reply:
Yes, of course. I hadn't thought about that when I changed the logic to no longer find chords.